### PR TITLE
external/webserver: resolves the crash and init buf 

### PR DESCRIPTION
**Pull request by amitpurwar**
This resolves the crash due to tcb and initialize the buffer

### CODE REVIEW

File: os/net/lwip/src/api/sockets.c [hunk 1]
```diff
@@ -388,7 +388,12 @@ struct lwip_sock *get_socket(int s, pid_t pid)
 	}
 
 	tcb = sched_gettcb(pid);
-	DEBUGASSERT(tcb && tcb->group);
+	if (tcb == NULL) {
```
Comment (pillip8282) [inline os/net/lwip/src/api/sockets.c:391]: I think that if tcb is null then system is corrupted. So it should be stop.

Comment (davidfather) [reply os/net/lwip/src/api/sockets.c:391]: @amitpurwar I don't understand why you are trying to add this kind of strange code.

**Review by davidfather — CHANGES_REQUESTED** [review]
I think this debug code should be on your local PC. What do you say?

File: os/net/lwip/src/api/sockets.c (hunk 1 above)
Comment (amitpurwar) [reply os/net/lwip/src/api/sockets.c:391]: we are analyzing the issue further and update the code accordingly

File: external/webserver/http.c [hunk 2]
```diff
@@ -98,6 +98,39 @@ pthread_addr_t http_server_handler(pthread_addr_t arg)
 	struct sockaddr_in client_addr;
 	struct mq_attr mqattr;
 	struct http_server_t *server = (struct http_server_t *)arg;
+	int reuse = 1;
+
+
```
Comment (pillip8282) [inline external/webserver/http.c:103]: please remove additional line

**Review by pillip8282 — COMMENTED** [review]
Could you explain the patch in more detail. I couldn't understand it.

**Review by pillip8282 — COMMENTED** [review]
I have questions about commit message "external/werbserver: Resolve the webserver crash"
commit message explained that connection accept thread(http_server_handler) called accept, and inside of accept it calls get_socket with terminate pid. However lwip_accept calls get_socket with getpid(). Then it pass the pid of http_server_handler not httptest_cb. So It doesn't access termitated pid(httptest_cb). Please let me know if I missed something.

**Comment by amitpurwar** [comment]
> getpid(

**Comment by amitpurwar** [comment]
> getpid(

**Comment by amitpurwar** [comment]
when I tested, getpid() from lwip_accept give PID of httptest_cb

**Review by pillip8282 — APPROVED** [review]
Thanks for explanation

**Review by pillip8282 — APPROVED** [review]

**Comment by amitpurwar** [comment]
when I tested, getpid() from lwip_accept give PID of httptest_cb

> [davidfather](/davidfather)

@davidfather  : Can you please review

**Review by davidfather — APPROVED** [review]
lgtm.

**Pull request merged**